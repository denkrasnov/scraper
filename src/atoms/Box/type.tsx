import { ReactNode } from "react";

import { Color } from "../Colors/types";

type Display =
  | "block"
  | "inline-block"
  | "inline"
  | "flex"
  | "inline-flex"
  | "none";

type FlexDirection = "column" | "row";

type JustifyContent =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "space-evenly";

export type AlignItems =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch";

type FlexWrap = "wrap" | "nowrap";

export interface BoxProps {
  alignItems?: AlignItems;
  backgroundColor?: Color;
  children: ReactNode;
  display?: Display;
  flexDirection?: FlexDirection;
  flexWrap?: FlexWrap;
  grow?: number;
  justifyContent?: JustifyContent;
  padding?: string;
  margin?: string;
  width?: string;
  height?: string;
  maxWidth?: string;
}
