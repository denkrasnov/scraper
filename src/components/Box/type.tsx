import { ReactNode } from "react";

type Display =
  | "block"
  | "inline-block"
  | "inline"
  | "flex"
  | "inline-flex"
  | "none";

type FlexDirection = "column" | "row";

type JustifyContent =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "space-evenly";

export type AlignItems =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch";

type FlexWrap = "wrap";

export interface BoxProps {
  alignItems?: AlignItems;
  children: ReactNode;
  display?: Display;
  flexDirection?: FlexDirection;
  flexWrap?: FlexWrap;
  justifyContent?: JustifyContent;
  grow?: number;
  width?: string;
  height?: string;
}
