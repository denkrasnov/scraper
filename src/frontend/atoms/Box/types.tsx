import { ReactNode } from "react";

import { Color } from "../Colors/types";
import { Space } from "../types";

type Display = "block" | "inlineBlock" | "inline" | "flex" | "inlineFlex";

type FlexDirection = "column";

type JustifyContent =
  | "flexEnd"
  | "center"
  | "spaceBetween"
  | "spaceAround"
  | "spaceEvenly";

export type AlignItems =
  | "flexStart"
  | "flexEnd"
  | "center"
  | "baseline"
  | "stretch";

type FlexWrap = "wrap" | "nowrap";

type Cursor = "pointer";

type FlexGrow = "gr1";

export interface BoxProps {
  alignItems?: AlignItems;
  background?: Color;
  children?: ReactNode;
  display?: Display;
  flexDirection?: FlexDirection;
  flexWrap?: FlexWrap;
  flexGrow?: FlexGrow;
  justifyContent?: JustifyContent;
  padding?: Space;
  margin?: Space;
  marginTop?: Space;
  marginBottom?: Space;
  marginRight?: Space;
  width?: string;
  height?: string;
  minHeight?: string;
  maxWidth?: string;
  cursor?: Cursor;
  onClick?: () => void;
}
