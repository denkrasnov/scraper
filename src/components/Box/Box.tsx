import React, { FC } from "react";

import { BoxProps } from "./type";
import styles from "./Box.css";

const Box: FC<BoxProps> = props => {
  const { children, ...rest } = props;
  return (
    <div className={styles.Box} style={{ ...rest }}>
      {children}
    </div>
  );
};

export default Box;
