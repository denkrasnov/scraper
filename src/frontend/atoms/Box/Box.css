@import "../Colors/backgroundColor.css";
@import "../helpers/space.css";

.root {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

.display-block {
  display: block;
}

.display-inlineBlock {
  display: inline-block;
}

.display-inline {
  display: inline;
}

.display-inlineFlex {
  display: inline-flex;
}

.justifyContent-flexEnd {
  justify-content: flex-end;
}

.justifyContent-center {
  justify-content: center;
}

.justifyContent-spaceBetween {
  justify-content: space-between;
}

.justifyContent-spaceAround {
  justify-content: space-around;
}

.justifyContent-spaceEvenly {
  justify-content: space-evenly;
}

.flexDirection-column {
  flex-direction: column;
}

.alignItems-flexStart {
  align-items: flex-start;
}

.alignItems-flexEnd {
  align-items: flex-end;
}

.alignItems-center {
  align-items: center;
}

.alignItems-baseline {
  align-items: baseline;
}

.alignItems-stretch {
  align-items: stretch;
}

.flexWrap-wrap {
  flex-wrap: wrap;
}

.flexWrap-nowrap {
  flex-wrap: nowrap;
}

.flexGrow-gr1 {
  flex-grow: 1;
}

.cursor-pointer {
  cursor: pointer;
}
