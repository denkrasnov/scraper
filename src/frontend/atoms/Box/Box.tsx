import React, { FC } from "react";

import mapPropsToStyles from "../helpers/mapPropsToStyles";
import { StyleType } from "../types";
import { BoxProps } from "./types";
import styles from "./Box.css";

const Box: FC<BoxProps> = (props) => {
  const {
    children,
    onClick,
    width,
    height,
    minHeight,
    maxWidth,
    ...restProps
  } = props;

  const classes = mapPropsToStyles(restProps, styles);

  const style: StyleType = {};

  if (width) style.width = width;
  if (height) style.height = height;
  if (minHeight) style.minHeight = minHeight;
  if (maxWidth) style.maxWidth = maxWidth;

  return (
    <div className={classes} onClick={onClick} style={style}>
      {children}
    </div>
  );
};

export default Box;
