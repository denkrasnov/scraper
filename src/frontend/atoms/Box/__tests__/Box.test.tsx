import { shallow } from "enzyme";

import Box from "../Box";

describe("Box", () => {
  const children = <div />;

  it("should render", () => {
    const component = shallow(<Box>{children}</Box>);
    expect(component).toMatchSnapshot();
  });

  it("should render with attributes", () => {
    const component = shallow(
      <Box
        background="BACKGROUND"
        bottom="10px"
        height="10px"
        maxWidth="10px"
        minHeight="10px"
        width="10px"
      >
        {children}
      </Box>
    );
    expect(component).toMatchSnapshot();
  });
});
