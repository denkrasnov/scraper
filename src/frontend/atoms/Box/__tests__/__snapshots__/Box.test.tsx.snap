// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render 1`] = `
<div
  className="root"
  style={Object {}}
>
  <div />
</div>
`;

exports[`Box should render with attributes 1`] = `
<div
  className="root background-BACKGROUND"
  style={
    Object {
      "height": "10px",
      "maxWidth": "10px",
      "minHeight": "10px",
      "width": "10px",
    }
  }
>
  <div />
</div>
`;
